src/pytorch_lightning/strategies/strategy.py

```diff
@@ -170,6 +170,16 @@ def optimizer_state(self, optimizer: Optimizer) -> Dict[str, Tensor]:
 
         Allows for syncing/collating optimizer state from processes in custom plugins.
         """
+        if isinstance(optimizer, LightningOptimizer):
+            optimizer = optimizer._optimizer
+
+        if hasattr(optimizer, "consolidate_state_dict"):
+            # there are optimizers like Fairscale's OSS or PyTorch's ZeroRedundancyOptimizer that shard their
+            # states, and to avoid OOM we consolidate the full state on rank 0 only
+            optimizer.consolidate_state_dict()
+            return optimizer.state_dict() if self.is_global_zero else {}
```

Comment:
In the deleted code, `rank_zero_only` was used which checks for the local rank, whereas this checks for the global rank. Is this intended?

Reply:
`rank_zero_only` means global rank in general. 
The special case is, when running on single node, `local_rank == global_rank`
Does that answer your question?

We could also decorate with rank_zero_only here, but the problem with that is it returns None on rank > 0. But I think it makes way more sense to return the empty dict.